extensions/amp-consent/0.1/consent-policy-manager.js

```diff
@@ -127,13 +119,34 @@ export class ConsentPolicyInstance {
     /** @private {!Array<string>} */
     this.pendingItems_ = pendingItems;
 
-    /** @private {?function()} */
+    /** @private {!Object<string, CONSENT_ITEM_STATE>} */
+    this.itemMap_ = map();
+
+    /** @private {number} */
+    this.pendingItemCount_ = 0;
+
+    /** @private {number} */
+    this.rejectedItemCount_ = 0;
+
+    /** @private {?function(CONSENT_POLICY_STATE)} */
     this.readyPromiseResolver_ = null;
 
-    /** @private {!Promise} */
+    /** @private {!Promise<CONSENT_POLICY_STATE>} */
     this.readyPromise_ = new Promise(resolve => {
       this.readyPromiseResolver_ = resolve;
     });
+
+    this.init_(pendingItems);
+  }
+
+  /**
+   * @param {!Array<string>} pendingItems
+   */
+  init_(pendingItems) {
+    for (let i = 0; i < pendingItems.length; i++) {
+      this.itemMap_[pendingItems[i]] = CONSENT_ITEM_STATE.UNKNOWN;
+    }
+    this.pendingItemCount_ = Object.keys(this.itemMap_).length;
```

Comment:
`this.pendingItemCount_ = pendingItems.length`?

Reply:
can't do that. We never check the uniqueness of items  in pendingItems. A good use case for you to ask for Set polyfill : )

Reply:
arggggh. 🙃